### ISSUE TITLE
Strings missing on translate.peeringdb.com

### ISSUE THREAD

**Issue by gordon-shumway-net** [peeringdb/peeringdb#714]
Hey together,

first the question for future topics in regard to the translation.
Is the preferred way to talk about such things on the [Mailinglist,](https://lists.peeringdb.com/cgi-bin/mailman/listinfo/pdb-translate) or here on GitHub?

Now to the main point of that Issue, I couldn't find another Issue which talks about that, so a new one now.

There are some strings who doesn't seem to be on translate.peeringdb.com as they are in English in in the most translated languages (Korean, Romania, German). Also in the search I'm not able to find them.
Two examples and the only I found yet are:

On the Beta Main Site the warning:
> automatically scheduled refreshes from production are currently disabled. 

On the Profile Page the last explanation in regard to the Affiliation:
> Affiliation is your own organisation. That means you are an employee or otherwise working for the organisation. The affiliation request is sent to support@peeringdb.com if no one is affiliated to the organisation yet. Otherwise, the admins are contacted. Only affiliate once per organisation. If you don't get an answer within a couple of days, please contact support@peeringdb.com.

Other strings I would post here as well if it is fine. :)

**Comment by arnoldnipper** [comment]
> first the question for future topics in regard to the translation.
> Is the preferred way to talk about such things on the [Mailinglist,](https://lists.peeringdb.com/cgi-bin/mailman/listinfo/pdb-translate) or here on GitHub?

Depends on whether it touches translate specific questions which should go the ml or technical which should go here.



> On the Beta Main Site the warning:
> 
> > automatically scheduled refreshes from production are currently disabled.

This is a string which only shows up on beta. Hence it is not translatable. 



> On the Profile Page the last explanation in regard to the Affiliation:

That definitely should show up. @ccaputo, would you mind to check?

**Comment by gordon-shumway-net** [comment]
> > first the question for future topics in regard to the translation.
> > Is the preferred way to talk about such things on the [Mailinglist,](https://lists.peeringdb.com/cgi-bin/mailman/listinfo/pdb-translate) or here on GitHub?
> 
> Depends on whether it touches translate specific questions which should go the ml or technical which should go here.

Okay, so things like that topic we have here are actually more for GitHub?
There is another topic I would like to talk about, it's the same kind of topic, but to different to put it in that Issue here.

> > On the Beta Main Site the warning:
> > > automatically scheduled refreshes from production are currently disabled.
> 
> This is a string which only shows up on beta. Hence it is not translatable.

Okay, fine!

**Comment by ccaputo** [comment]
This stumped me today.  I upgraded weblate from 3.6.1 to 4.0.4 and it didn't help.  Have opened a bug with weblate to hopefully get help: https://github.com/WeblateOrg/weblate/issues/3847

**Comment by gordon-shumway-net** [comment]
So you @ccaputo put new strings into Weblate and the only Language which doesn't accept them is Korean?
All other Languages have new Strings now. Am I see that right?

**Comment by ccaputo** [comment]
Pretty much.  Korean is wedged for some reason.

**Comment by ccaputo** [comment]
I believe this is now fixed.  For future reference in case we hit this again, this was fixed on the server, as user weblate, with:

```
(cd /srv/translate.peeringdb.com && DJANGO_SETTINGS_MODULE=main_conf.settings /srv/translate.peeringdb.com/venv/bin/weblate loadpo --foreground --all --force)
```

If you confirm things look good, feel free to close this.  Thanks!

**Comment by gordon-shumway-net** [comment]
Great to hear that!
Thank you for your help @ccaputo 
Looks great so far :)

**Comment by gordon-shumway-net** [comment]
Hey together,
I could find some other strings (behind the second dash is the location where it appears)who seem to doesn't appear on Weblate and because of that, appear in English on the Korean version of beta.peerigndb.com. Can someone maybe check that?

- Users requesting affiliation
- Users in Organization
- Policy
- Private Peering Facilities
- Local Facilities
- Filter 
- Present
- Absent
- Chinese (Simplified)
- Chinese (Traditional)

Thank you! :)

**Comment by ccaputo** [comment]
Most of the strings listed as not being translated are in https://github.com/peeringdb/peeringdb/tree/master/peeringdb_server/templates/site and fail to have a `{% trans "` and `" %}` wrapper for some reason, causing them to not be flagged for translation.  For example "Users requesting affiliation" is in https://github.com/peeringdb/peeringdb/blob/master/peeringdb_server/templates/site/view_organization_tools.html as:

` <h5>Users requesting affiliation</h5>`

rather than:

` <h5>{% trans "Users requesting affiliation" %}</h5>`

@grizz and @vegu Are these oversights or is there a reason they aren't marked for translation?  Also, I wasn't able to figure out why the "Chinese" language names aren't being offered up for translation by "django-admin makemessages" while other language names are.  Thoughts on that?  Of note is that on beta, "Occitan" shows up as "oc" in the drop-down for changing languages.

**Comment by gordon-shumway-net** [comment]
Hey together,
we found some additonal strings who seem to be missed on Weblate.

* 0-20Mbps
* 20-50Gbps
* 50-100Gbps
* 100-200Gbps
* 200-300Gbps
* 300-500Gbps
* 500-1000Gbps
* 1-5Tbps
* 5-10Tbps
* 10-20Tbps
* 20-50Tbps
* 50-100Tbps
* 100+ Tbps
* captcha
* dubai
* Your applications
* No applications defined. Click here if you want to register a new one
* And everything else on the "Register a new application" Page (https://beta.peeringdb.com/oauth2/applications/register/)

Thank you! :)

**Comment by ccaputo** [comment]
Likely needs fix: https://github.com/peeringdb/peeringdb/blob/69de96dbe9427863a9ece2f9125a9197396f5795/peeringdb_server/templates/site/view_network_side.html#L190

**Comment by grizz** [comment]
> @grizz and @vegu Are these oversights or is there a reason they aren't marked for translation?

Those look like oversights to me.

> Also, I wasn't able to figure out why the "Chinese" language names aren't being offered up for translation by "django-admin makemessages" while other language names are. Thoughts on that?

None without looking further into it, that's strange.

> Of note is that on beta, "Occitan" shows up as "oc" in the drop-down for changing languages.

If it's not in django.global_settings, it needs a manual mapping added at https://github.com/peeringdb/peeringdb/blob/69de96dbe9427863a9ece2f9125a9197396f5795/mainsite/settings/__init__.py#L564

**Comment by ccaputo** [comment]
> > Also, I wasn't able to figure out why the "Chinese" language names aren't being offered up for translation by "django-admin makemessages" while other language names are. Thoughts on that?
> 
> None without looking further into it, that's strange.

If you can help, that would be great.

> > Of note is that on beta, "Occitan" shows up as "oc" in the drop-down for changing languages.
> 
> If it's not in django.global_settings, it needs a manual mapping added at
> 
> https://github.com/peeringdb/peeringdb/blob/69de96dbe9427863a9ece2f9125a9197396f5795/mainsite/settings/__init__.py#L564

Ok, I submitted a PR for Occitan.  I trust it will only show up on beta and that production choice of languages is controlled elsewhere?

Also, I am confused why English shows up in the dropdown twice on beta and once on production.

**Comment by gordon-shumway-net** [comment]
Like @ccaputo mentioned in issue #725 I created an Pull Request for all in Weblate missing Strings I was able to find in the codebase. #731 
Also some additional, not yet mentioned in this Issue:
* The complete AUP site (https://peeringdb.com/aup)
* The complete Org Merge Tools Site (https://github.com/peeringdb/peeringdb/blob/master/peeringdb_server/templates/admin/org_merge_tool.html)

There are some more strings where it would be cool if you guys could check what's going on.
* 0-20Mbps
* 20-50Gbps
* 50-100Gbps
* 100-200Gbps
* 200-300Gbps
* 300-500Gbps
* 500-1000Gbps
* 1-5Tbps
* 5-10Tbps
* 10-20Tbps
* 20-50Tbps
* 50-100Tbps
* 100Tbps
_Explanation: This strings appear in the Advanced Search for Networks in the Traffic Levels Filter. Also it appears if you wanna change the Traffic Level for a network. Maybe it has something to do with pull request #723_
* The complete Manage OAuth2 Applications Page
_Explanation: this site: https://beta.peeringdb.com/oauth2/applications/ and this site: https://beta.peeringdb.com/oauth2/applications/register/_

To check it on the website, the best would be to change the language to Korean, because then it's obvious and more easy to recognize non translated strings. :)

**Comment by grizz** [comment]
> > > Also, I wasn't able to figure out why the "Chinese" language names aren't being offered up for translation by "django-admin makemessages" while other language names are. Thoughts on that?
> > 
> > 
> > None without looking further into it, that's strange.
> 
> If you can help, that would be great.

Ok, could you give more information? After re-reading it, I'm not sure exactly what you mean. Should probably be another issue as well.

> > > Of note is that on beta, "Occitan" shows up as "oc" in the drop-down for changing languages.
> > 
> > 
> > If it's not in django.global_settings, it needs a manual mapping added at
> > https://github.com/peeringdb/peeringdb/blob/69de96dbe9427863a9ece2f9125a9197396f5795/mainsite/settings/__init__.py#L564
> 
> Ok, I submitted a PR for Occitan. I trust it will only show up on beta and that production choice of languages is controlled elsewhere?

We can just add it in that settings file, no need for it to wait on them to PR.

> Also, I am confused why English shows up in the dropdown twice on beta and once on production.

That sounds like a bug. :)

**Comment by ccaputo** [comment]
> > > > Also, I wasn't able to figure out why the "Chinese" language names aren't being offered up for translation by "django-admin makemessages" while other language names are. Thoughts on that?
> > > 
> > > None without looking further into it, that's strange.
> > 
> > If you can help, that would be great.
> 
> Ok, could you give more information? After re-reading it, I'm not sure exactly what you mean. Should probably be another issue as well.

Please see this screenshot.  Both Chinese language names are not available for translation.

<img width="159" alt="Screen Shot 2020-05-27 at 1 25 59 PM" src="https://user-images.githubusercontent.com/1417833/83068797-b6d30580-a01d-11ea-8258-bb05591daae5.png">

> > > > Of note is that on beta, "Occitan" shows up as "oc" in the drop-down for changing languages.
> > > 
> > > If it's not in django.global_settings, it needs a manual mapping added at
> > > https://github.com/peeringdb/peeringdb/blob/69de96dbe9427863a9ece2f9125a9197396f5795/mainsite/settings/__init__.py#L564
> > 
> > Ok, I submitted a PR for Occitan. I trust it will only show up on beta and that production choice of languages is controlled elsewhere?
> 
> We can just add it in that settings file, no need for it to wait on them to PR.

Into the deployed containers?  I am not clear on how this would be done.

**Comment by ccaputo** [comment]
> > > > > Also, I wasn't able to figure out why the "Chinese" language names aren't being offered up for translation by "django-admin makemessages" while other language names are. Thoughts on that?
> > > > 
> > > > None without looking further into it, that's strange.
> > > 
> > > If you can help, that would be great.
> > 
> > Ok, could you give more information? After re-reading it, I'm not sure exactly what you mean. Should probably be another issue as well.
> 
> Please see this screenshot. Both Chinese language names are not available for translation.
> 
> <img alt="Screen Shot 2020-05-27 at 1 25 59 PM" width="159" src="https://user-images.githubusercontent.com/1417833/83068797-b6d30580-a01d-11ea-8258-bb05591daae5.png">

Fixed this part with:

https://github.com/peeringdb/translations/commit/b876f88789e9d4b2110eb4033930daf999ed0e0d

**Comment by gordon-shumway-net** [comment]
> Fixed this part with:
> 
> [peeringdb/translations@b876f88](https://github.com/peeringdb/translations/commit/b876f88789e9d4b2110eb4033930daf999ed0e0d)

Okay cool, that brought a bunch (maybe 20) of new strings to Weblate!

**Comment by grizz** [comment]
I think all the issues here are fixed.

**Comment by gordon-shumway-net** [comment]
Hey @grizz ,
what about that?
Is that fixed? I don't think so... or am I wrong?

> There are some more strings where it would be cool if you guys could check what's going on.
> 
>     * 0-20Mbps
> 
>     * 20-50Gbps
> 
>     * 50-100Gbps
> 
>     * 100-200Gbps
> 
>     * 200-300Gbps
> 
>     * 300-500Gbps
> 
>     * 500-1000Gbps
> 
>     * 1-5Tbps
> 
>     * 5-10Tbps
> 
>     * 10-20Tbps
> 
>     * 20-50Tbps
> 
>     * 50-100Tbps
> 
>     * 100Tbps
>       _Explanation: This strings appear in the Advanced Search for Networks in the Traffic Levels Filter. Also it appears if you wanna change the Traffic Level for a network. Maybe it has something to do with pull request #723_
> 
>     * The complete Manage OAuth2 Applications Page
>       _Explanation: this site: https://beta.peeringdb.com/oauth2/applications/ and this site: https://beta.peeringdb.com/oauth2/applications/register/_
> 
> 
> To check it on the website, the best would be to change the language to Korean, because then it's obvious and more easy to recognize non translated strings. :)

**Comment by ccaputo** [comment]
> Hey @grizz ,
> what about that?
> Is that fixed? I don't think so... or am I wrong?
> 
> > There are some more strings where it would be cool if you guys could check what's going on.
> > ```
> > * 0-20Mbps
> > * 20-50Gbps
> > * 50-100Gbps
> > * 100-200Gbps
> > * 200-300Gbps
> > * 300-500Gbps
> > * 500-1000Gbps
> > * 1-5Tbps
> > * 5-10Tbps
> > * 10-20Tbps
> > * 20-50Tbps
> > * 50-100Tbps
> > * 100Tbps
> >   _Explanation: This strings appear in the Advanced Search for Networks in the Traffic Levels Filter. Also it appears if you wanna change the Traffic Level for a network. Maybe it has something to do with pull request #723_```

Per https://github.com/peeringdb/peeringdb/issues/519#issuecomment-612924553 there is some complication with this in that there is a coordinated update with django-peeringdb.  Beta is currently using a zip archive of a django-peeringdb branch (https://github.com/peeringdb/peeringdb/blob/7ee0fa73f9d175e79561059959196ccc4f305e43/Pipfile#L25) while the translations software is using the django-peeringdb master branch.  Until the django-branch https://github.com/peeringdb/django-peeringdb/compare/gh_pdb_519 is merged into master, the translations software will not have the new strings and so they won't be available to be translated.  @grizz when may the branch be merged to master?

> > ```* The complete Manage OAuth2 Applications Page
> >   _Explanation: this site: https://beta.peeringdb.com/oauth2/applications/ and this site: https://beta.peeringdb.com/oauth2/applications/register/_
> > ```

Figured this one out!  We are now including django-oauth-toolkit as part of the list of projects to translate.   You should see new strings relating to it.  Thanks!  (https://github.com/peeringdb/translations/commit/028d2da558c4c7fb505e1588fa19eaafc40f87ad)

**Comment by grizz** [comment]
Oh, I see the other comment for missed strings now, will re-open. @gordon-shumway-net please make new issues for additional strings you find in the future, it's hard to track what's done and what isn't otherwise. :)

> @grizz when may the branch be merged to master?

Definitely before next release, we were waiting on the rest of the issues for the next release to be done. Once it's merged, anyone using master will not be able to sync until the production release.

**Comment by grizz** [comment]
Added #752 for the remaining strings.

Traffic speeds added with #519
